articles/introduction-to-r/index.md

```diff
@@ -0,0 +1,328 @@
+
+```r
+new_matrix <- matrix(1:4, byrow=TRUE, ncol=2)
+colnames(newm_matrix) <- c("column 1","column 2","column 3")
+rownames(newm_matrix) <- c("row 1","row 2","row 3")
+```
+##### Summing Entries
+
+Let's say you want to compute the average of the entries in your matrix. We begin by adding them up and then dividing by the total number of entries. R offers functions to add the entire rows or columns. They are `rowSums,` `colSums.` These functions take the matrix as their input and output a vector containing the sums.
+
+Example:
+```r
+x <- matrix(1:9, byrow = TRUE, nrow=3)
+row_sum <- rowSums(x)
+col_sum <- colSums(x)
+```
+
+##### Inserting new rows and columns to existing matrices
+
+Let us consider the situation where we need to modify a matrix by adding extra rows or columns. R provides two functions called `cbind()` and `rbind()` to do the same.
+The input to the function is the matrix to be modified followed by the vector to be inserted. Consider the example given below:
+```r
+x <- matrix(1:9, byrow = TRUE, nrow=3)
+new_matrix_with_additional_column <- cbind(x, c(1,2,3))
+new_matrix_with_additional_row <- rbind(new_matrix_with_additional_column, c(4,5,6))
+new_matrix_with_additional_row
+```
+The ouput for the above code is given as follows:
+```txt
+ new_matrix_with_additional_row
+     [,1] [,2] [,3] [,4]
+[1,]    1    2    3    1
+[2,]    4    5    6    2
+[3,]    7    8    9    3
+[4,]    4    5    6    4
+```
+
+##### Selecting Elements
+The final concept to be covered under matrices is the selection of elements. Matrices in R work like 2-D arrays in other programming languages. Therefore all the slicing operations stand true in R. Let us assume we have a `4X4` dimensional matrix. The first quarter is a `2X2` matrix. To get the first quarter of the matrix, we will use the following code:
+
+```r
+x <- matrix(1:16, brow=TRUE, nrow=4) # matrix definition
+x[1:2,1:2] # selection operation
+```
+The output is given as follows:
+```txt
+x
+     [,1] [,2] [,3] [,4]
+[1,]    1    2    3    4
+[2,]    5    6    7    8
+[3,]    9   10   11   12
+[4,]   13   14   15   16
+
+ x[1:2,1:2]
+     [,1] [,2]
+[1,]    1    2
+[2,]    5    6
+```
+
+With the solid knowledge of vectors, matrices, and various data types, let's shift gears and move towards factors and lists.
+
+#### Factors
+In the case of supervised learning algorithms, we use factors. [classification](https://www.section.io/engineering-education/supervised-learning-algorithms/) algorithms have discrete output variables, whereas regression algorithms have continuous output variables. We can use vectors and matrices to store continuous output variables. But in the case of categorical values, we use factors.
+
+R provides us with the function factor to deal with categorical values. Examples are the best way to understand the implementation.
+
+```r
+factor_vector <- c("Big","Small","Medium")
+factored_vector <- factor(factor_vector)
```

Comment:
Needs comment explaining it turns the vector into a factor, and what this actually means